atom/common/promise_util.cc

```diff
@@ -11,7 +11,6 @@ namespace atom {
 namespace util {
 
 Promise::Promise(v8::Isolate* isolate) {
-  DCHECK_CURRENTLY_ON(content::BrowserThread::UI);
```

Comment:
Why is this a bad DCHECK? Don't promises need to be created on the UI thread?

Reply:
It was failing even when created on the UI thread.  Turns out you can't DCHECK a thread when the app isn't ready.